firefox-ios/Client/Frontend/Settings/AppIconSelection/AppIconView.swift

```diff
@@ -0,0 +1,84 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import SwiftUI
+import Common
+
+struct AppIconView: View, ThemeApplicable {
+    let appIcon: AppIcon
+    let isSelected: Bool
+    let windowUUID: WindowUUID
+
+    let setAppIcon: (AppIcon) -> Void
+
+    // MARK: - Theming
+    // FIXME FXIOS-11472 Improve our SwiftUI theming
+    @Environment(\.themeManager)
+    var themeManager
+    @State private var themeColors: ThemeColourPalette = LightTheme().colors
```

Comment:
This is nice if you don't want to use state var for each color 🚀 
Only confusing part is `LightTheme().colors` so if somehow we can have something like Theme().colors to get the current theme then this would make it so much better but its more of a nit for me


    ```@State private var themeColors: ThemeColourPalette = LightTheme().colors```